### PR TITLE
Adds movement and direction switch for scopes with ctrl+Wasd

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Scopes now let you while using them, However they incur heavy slowdown. you can also now use ctrl+wasd to spin around and change what way youre scoping to.

## Why It's Good For The Game
it gives scopes a bit more of space outside only on snipers and removes lots of annoyance to them.

## Changelog
:cl:
balance: scopes now incur slowdown instead of unscoping when moving away.
qol: you can now change which direction you're scoping with ctrl+wasd(or whatever you use to face without moving).
/:cl:

### CODE REVIEW

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -129,6 +129,7 @@
 #define COMSIG_ATOM_EXITED "atom_exited"						//from base of atom/Exited(): (atom/movable/exiting, direction)
 #define COMSIG_ATOM_BUMPED "atom_bumped"						///from base of atom/Bumped(): (/atom/movable)
 #define COMSIG_ATOM_DIR_CHANGE "atom_dir_change"				//from base of atom/setDir(): (old_dir, new_dir)
+	#define COMSIG_ATOM_DIR_CHANGE_BLOCKED (1<<0)
```
Comment (SplinterGP) [inline code/__DEFINES/dcs/signals.dm:132]:
```suggestion
```

**Comment by Alexeivichi** [comment]
Tactical aiming lets go!!!

File: code/__DEFINES/dcs/signals.dm [hunk 2]
```diff
@@ -334,6 +334,7 @@
 #define COMSIG_MOB_GUN_AUTOFIRED "mob_gun_autofired"
 #define COMSIG_MOB_ATTACHMENT_FIRED "mob_attachment_fired"
 #define COMSIG_MOB_TOGGLEMOVEINTENT "mob_togglemoveintent"		//drom base of mob/toggle_move_intent(): (new_intent)
+#define COMSIG_MOB_FACE_DIR "mob_face_dir"
```
Comment (TiviPlus) [inline code/__DEFINES/dcs/signals.dm:337]: delete

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -715,30 +727,38 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
...
-				user.client.pixel_x = -viewoffset
-				user.client.pixel_y = 0
 
 	user.visible_message(span_notice("[user] peers through \the [zoom_device]."),
 	span_notice("You peer through \the [zoom_device]."))
 	zoom = TRUE
 	RegisterSignal(user, COMSIG_ITEM_ZOOM, .proc/zoom_check_return)
 	onzoom(user)
 
+/obj/item/proc/change_zoom_offset(mob/user, newdir, zoom_offset)
```
Comment (TiviPlus) [inline code/game/objects/items.dm:739]: Autodoc

File: code/modules/mob/mob.dm [hunk 4]
```diff
@@ -587,6 +587,7 @@
 /mob/proc/facedir(ndir)
 	if(!canface())
 		return FALSE
+	SEND_SIGNAL(src, COMSIG_MOB_FACE_DIR, ndir)
```
Comment (TiviPlus) [inline code/modules/mob/mob.dm:590]: delete

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob.dm (hunk 4 above)
Comment (SplinterGP) [reply code/modules/mob/mob.dm:590]: why ? this is used so it is player inputted to change the dir and not changed by everything else like *spin and other stuff, causing people to *vomit

File: code/game/objects/items.dm (hunk 3 above)
Comment (SplinterGP) [inline code/game/objects/items.dm:739]:
```suggestion
///applies the offset of the zooming, using animate for smoothing.
/obj/item/proc/change_zoom_offset(mob/user, newdir, zoom_offset)
```

File: code/modules/mob/mob.dm (hunk 4 above)
Comment (TiviPlus) [reply code/modules/mob/mob.dm:590]: I mean if tehy spin with a scope and make themselves vomit sounds like a personal issue tbh

**Comment by SplinterGP** [comment]
gotta fix

**Comment by SplinterGP** [comment]
scopes are still bugged btw, scopes that arent supposed to increase view are increasing like miniscope, but thats not a fault of mine, it was #6748 that broke it

File: code/modules/projectiles/gun_attachables.dm [hunk 5]
```diff
@@ -742,12 +754,14 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	wield_delay_mod = 0.2 SECONDS
 	accuracy_mod = 0.05
 	accuracy_unwielded_mod = -0.05
-	aim_speed_mod = 0.04 SECONDS
-	zoom_offset = 5
-	zoom_viewsize = 7
+	aim_speed_mod = 0.04
```
Comment (BraveMole) [inline code/modules/projectiles/gun_attachables.dm:757]:
```suggestion
	aim_speed_mod = 0.5
```

File: code/modules/projectiles/gun_attachables.dm [hunk 6]
```diff
@@ -612,18 +612,19 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	icon_state = "sniperscope"
 	desc = "A rail mounted zoom sight scope. Allows zoom by activating the attachment. Use F12 if your HUD doesn't come back."
 	slot = ATTACHMENT_SLOT_RAIL
-	aim_speed_mod = 0.06 SECONDS //Extra slowdown when aiming
+	aim_speed_mod = 0.06 //Extra slowdown when aiming
```
Comment (BraveMole) [inline code/modules/projectiles/gun_attachables.dm:615]:
```suggestion
	aim_speed_mod = 0.5 //Extra slowdown when aiming
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_attachables.dm [hunk 7]
```diff
@@ -612,18 +612,19 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
...
-	///how many tiles to shift the users viewpoint
-	var/zoom_offset = 11
-	///how many tiles to increase the users view box
-	var/zoom_viewsize = 12
-	scoped_accuracy_mod = SCOPE_RAIL
+	zoom_tile_offset = 11
+	zoom_viewsize = 12
+	zoom_allow_movement = TRUE
+	///how much slowdown the scope gives when zoomed. You want this to be slowdown you want minus aim_speed_mod
+	var/zoom_slowdown = 1.44
```
Comment (SplinterGP) [inline code/modules/projectiles/gun_attachables.dm:627]:
```suggestion
	var/zoom_slowdown = 1
```

File: code/modules/projectiles/gun_attachables.dm (hunk 5 above)
Comment (SplinterGP) [inline code/modules/projectiles/gun_attachables.dm:757]:
```suggestion
	aim_speed_mod = 0.2
```

File: code/modules/projectiles/gun_attachables.dm [hunk 8]
```diff
@@ -742,12 +754,14 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
...
 	accuracy_unwielded_mod = -0.05
-	aim_speed_mod = 0.04 SECONDS
-	zoom_offset = 5
-	zoom_viewsize = 7
+	aim_speed_mod = 0.04
 	scoped_accuracy_mod = SCOPE_RAIL_MINI
 	scope_zoom_mod = TRUE
 	has_nightvision = FALSE
+	zoom_allow_movement = TRUE
+	zoom_slowdown = 0.46
```
Comment (SplinterGP) [inline code/modules/projectiles/gun_attachables.dm:762]:
```suggestion
	zoom_slowdown = 0.3
```

**Review by BraveMole — APPROVED** [review]

**Comment by MiniMeatwad** [comment]
Should remove the ability to move while scoped with the SADAR, since the SADAR can use a miniscope.

**Review by BraveMole — APPROVED** [review]

**Pull request merged**